data/plugins/downloaders/APODDownloader.py

```diff
@@ -0,0 +1,87 @@
+# -*- Mode: Python; coding: utf-8; indent-tabs-mode: nil; tab-width: 4 -*-
+### BEGIN LICENSE
+# Copyright (c) 2012, Peter Levi <peterlevi@peterlevi.com>
+# This program is free software: you can redistribute it and/or modify it
+# under the terms of the GNU General Public License version 3, as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranties of
+# MERCHANTABILITY, SATISFACTORY QUALITY, or FITNESS FOR A PARTICULAR
+# PURPOSE.  See the GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program.  If not, see <http://www.gnu.org/licenses/>.
+### END LICENSE
+
+import random
+import logging
+
+from variety import _
+from variety.Util import Util
+from variety.plugins.downloaders.SimpleDownloader import SimpleDownloader
+
+logger = logging.getLogger('variety')
+
+random.seed()
+
+
+class APODDownloader(SimpleDownloader):
+    DESCRIPTION = _("NASA Astro Pic of the Day")
+    ROOT_URL = "http://apod.nasa.gov/apod/"
+
+    @classmethod
+    def get_info(cls):
+        return {
+            "name": "APODDownloader",
+            "description": APODDownloader.DESCRIPTION,
+            "author": "Peter Levi",
+            "version": "0.1"
+        }
+
+    def get_description(self):
+        return APODDownloader.DESCRIPTION
```

Comment:
I think it would be nicer to use read-only properties instead of getters here: https://docs.python.org/3/library/functions.html#property. This way clients can skip the explicit function call and read the value as if it were an attribute.

It'd end up looking something like:

```python
@property
def description(self):
    return APODDownloader.DESCRIPTION
```

Reply:
I thought about this, but in this case it's more important that inheriting / implementing / overriding these is easy, not so much calling these, as we only have a limited number of calls of these that are rarely touched, while I'd like to enable contributors to implement image sources easily. And `@property` combined with inheritance and `abc.abstractmethod` makes things blurry as the overriding class must specify `@property` as well, otherwise the errors that arise are very confusing (if you override but forget the `@property` decoration, `x.description` will be a function instead of a string).

But in general, I am not a huge fan of `@property`, as it makes things less explicit for very little gain, and I prefer code to be explicit. 

I'm still thinking whether it makes sense to remove the `get_` prefixes. It sounds benefition for things like description, source_name, source_type. But several things bothers me: 

- Between `ImageSource` and `Downloader`, there is some overlap in these, and some are fields, while others are getters (e.g. `variety` and `get_variety`), and we have multiple inheritance between these in `SimpleDownloader`. So for example if we had the field `ImageSource.variety` and the method `Downloader.variety()`, this would cause issues (or confusion) in `SimpleDownloader`.

- Also, the getters provide me a with a proper location for documentation - what needs to be implemented (and how), what can be used as a helper, etc. E.g. if I remove `Downloader.get_config()` in favor of simply using `Downloader.config`, then I would need to move that documentation to `__init__`'s docstring.

- Additionally, `config` and `full_descriptor` are things that I like to be able to set in the constructor (so, for example once we have, say, `FlickrImageSource`, it will be able to create different `FlickrDownloader` instances with different `config`s), so it makes sense to have them as member fields. But I also would like to allow subclasses to implement them as computed properties and not use the member field, whenever this makes sense (e.g. `full_descriptor` might be a substring of `config` in some case). So having different naming for the accessor method and the field gives more flexibility in that aspect.

Reply:
Otherwise I agree that using `get_` prefixes is not a very Pythonic thing, it's more a Java approach, and makes things kinda ugly, but I do like the flexibility between "default, easy to set and use member field" vs "computed property" that this allows.

Once I implement `ConfigurableImageDownloader`, and then implement Flickr, Reddit, etc. as such, these abstract classes will probably change quite a bit, and once I have something ready and functional, I'll pass again and revise all this.